packages/contracts/contracts/foundry/Sphinx.sol

```diff
@@ -303,63 +340,53 @@ abstract contract Sphinx {
+        IGnosisSafe _safe,
+        address _moduleAddress
+    ) private returns (uint256[] memory) {
+        GnosisSafeTransaction[] memory txnArray = sphinxUtils.makeGnosisSafeTransactions(
+            address(_safe),
+            _accountAccesses
+        );
         uint256[] memory gasEstimates = new uint256[](txnArray.length);
 
-        // Deploy the Gnosis Safe if it's not already deployed. This is necessary because we're
-        // going to call the Gnosis Safe to estimate the gas.
-        if (address(safe).code.length == 0) {
-            // Deploy the Gnosis Safe and Sphinx Module. It's not strictly necessary to prank the
-            // Managed Service contract, but this replicates the prod environment for the DevOps
-            // Platform, so we do it anyways.
-            vm.startPrank(managedServiceAddress);
-            _sphinxDeployModuleAndGnosisSafe();
-            vm.stopPrank();
-        }
-
         // We prank the Sphinx Module to replicate the production environment. In prod, the Sphinx
         // Module calls the Gnosis Safe.
-        vm.startPrank(module);
+        vm.startPrank(_moduleAddress);
 
         for (uint256 i = 0; i < txnArray.length; i++) {
-            SphinxTransaction memory txn = txnArray[i];
+            GnosisSafeTransaction memory txn = txnArray[i];
             uint256 startGas = gasleft();
-            bool success = safe.execTransactionFromModule(
+            bool success = _safe.execTransactionFromModule(
                 txn.to,
                 txn.value,
                 txn.txData,
                 txn.operation
             );
-            gasEstimates[i] = startGas - gasleft();
+            uint256 finalGas = gasleft();
+
             require(success, "Sphinx: failed to call Gnosis Safe from Sphinx Module");
+
+            // Include a buffer to ensure the user's transaction doesn't fail on-chain due to
+            // variations between the simulation and the live execution environment. There are a
+            // couple areas in particular that could lead to variations:
+            // 1. The on-chain state could vary, which could impact the cost of execution. This is
+            //    inherently a source of variation because there's a delay between the simulation
+            //    and execution.
+            // 2. Foundry's simulation is treated as a single transaction, which means SLOADs are
+            //    more likely to be "warm" (i.e. cheaper) than the production environment, where
+            //    transactions may be split between batches.
+            //
+            // We chose to multiply the gas by 1.3 because multiplying it by a higher number could
+            // make a very large transaction unexecutable on-chain. Since the 1.3x multiplier
+            // doesn't impact small transactions very much, we add a constant amount of 20k too.
+            gasEstimates[i] = 20_000 + ((startGas - finalGas) * 13) / 10;
```

Comment:
This code was previously in the TypeScript `sphinxEstimateMerkleLeafGas` function, which no longer exists. I figured it makes the most sense to put it here.

Reply:
> Hypothetically, if this code did not work and we erroneously underestimated the gas cost for a transaction. Would that be caught during the current simulation step (which uses real providers)? Could that be a good reason to maintain the use of real providers in the simulation?

Great point. I added that situation to the [relevant ticket](https://linear.app/chugsplash/issue/CHU-479/investigate-how-to-make-the-simulation-more-reliable) for making the simulation more reliable.

> It's worth noting that an alternative way to cover this edge case could be with a simulation using real providers that happens in our backend.

Yeah, I think that would work. I'd need to hear more about the benefit of using our backend though, since I don't think I've encountered a situation where a public node provider underestimates the gas

Reply:
Oh, you meant that the Hardhat simulation would be more reliable with the real node providers in our backend, didn't you

Reply:
Yeah I just meant doing essentially the same thing we currently do in the proposal task, but handle it in our backend so the proposal is faster and more reliable.